### PR TITLE
feat: ctags completion source

### PR DESCRIPTION
**Pull request by sunjon**

### CODE REVIEW

File: lua/complementree/sources.lua [hunk 1]
```diff
@@ -162,17 +167,65 @@ local function lsp_completedone(completed_item)
...
+    ["m"] = "member",
+    ["p"] = "prototype",
+    ["s"] = "structure",
+    ["t"] = "typedef",
+    ["u"] = "union",
+    ["v"] = "variable",
+  },
+}
+
+M.ctags_matches = cached("ctags", function(_, _, _, _)
```
Comment (vigoux) [inline lua/complementree/sources.lua]: Tiny question: why not using `taglist()` here ?

File: lua/complementree/sources.lua [hunk 2]
```diff
@@ -215,6 +215,46 @@ local function lsp_completedone(completed_item)
...
+    ['u'] = 'union',
+    ['v'] = 'variable',
+  },
+}
+
+M.ctags_matches = cached('ctags', function(line_to_cursor, _)
+  local pref_start = line_to_cursor:find '%w*$'
+  local prefix = line_to_cursor:sub(pref_start)
+
+  local tags = vim.fn.taglist '.*'
```
Comment (vigoux) [inline lua/complementree/sources.lua]: Does `taglist ''` work here?

Comment (sunjon) [reply lua/complementree/sources.lua]: doesn't seem to

File: lua/complementree/sources.lua [hunk 3]
```diff
@@ -215,6 +215,46 @@ local function lsp_completedone(completed_item)
...
+
+  local filetype = 'default'
```
Comment (vigoux) [inline lua/complementree/sources.lua]: You make that configurable but don't retrieve the filetype value, maybe you should ?

File: lua/complementree/sources.lua [hunk 4]
```diff
@@ -215,6 +215,47 @@ local function lsp_completedone(completed_item)
...
+    ['v'] = 'variable',
+  },
+}
+
+M.ctags_matches = cached('ctags', function(line_to_cursor, _)
+  local pref_start = line_to_cursor:find '%w*$'
+  local prefix = line_to_cursor:sub(pref_start)
+
+  local filetype = vim.bo.filetype
+  filetype = ctags_extension.filetype or 'default'
```
Comment (vigoux) [inline lua/complementree/sources.lua]:
```suggestion
  local filetype = vim.bo.filetype
  local extensions = ctags_extension[filetype] or ctags_extension.default
```

File: lua/complementree/sources.lua [hunk 5]
```diff
@@ -215,6 +215,47 @@ local function lsp_completedone(completed_item)
...
+  local tags = vim.fn.taglist '.*'
+
+  local items = {}
+  for _, t in ipairs(tags) do
+    items[#items + 1] = {
+      word = t.name,
+      abbr = t.name,
+      kind = (t.kind and ctags_extension[filetype][t.kind] or 'undefined'),
```
Comment (vigoux) [inline lua/complementree/sources.lua]:
```suggestion
      kind = (t.kind and extensions[t.kind] or 'undefined'),
```

**Review by vigoux — COMMENTED** [review]
I think you'd like it this way rather than the current one.

**Pull request merged**